lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -2829,6 +2862,17 @@ bool QuakeBridgeVisitor::VisitCXXConstructExpr(clang::CXXConstructExpr *x) {
     return pushValue(builder.create<cc::LoadOp>(loc, copyObj));
   }
 
+  // Note we only screen for structs that contain data members.
+  // Revisit this if we ever enable Kernel structs with data members.
+  if (auto structTy = dyn_cast<cc::StructType>(ctorTy);
```

Comment:
Why are we generating constructors someplace other than when the constructor is visited?

Reply:
I originally constructed the full struct here, but for something like `MyStruct s(var0, var1, var1, ...)`, the result ended up with 2 struct creations and filling the struct. Not sure where the first one is created, but I figured you'd implemented it to handle construction somewhere already. Note that for `MyStruct s` (no constructor args), you are required to alloca here.